src/lib/packet.h

```diff
@@ -377,6 +377,7 @@ typedef enum {
                                * IETF-S/MIME) */
     PGP_PKA_EDDSA = 22,       /* EdDSA from draft-ietf-openpgp-rfc4880bis */
 
+    PGP_PKA_SM2_ENCRYPT = 98, /* SM2 encryption */
```

Comment:
I definitely get that and it makes sense, but I'm wondering if we should be thinking about this at a higher level. Like if I changed the question to: What would RFC 4880 9.1 actually look like if SM2 were added by the WG?

I get the impression that usage-specific PK algorithm identifiers are deprecated in the RFC in favor of key flags, so I'm not sure if they would add it separately like this.

From RFC 4880:

>    There are algorithm types for RSA Sign-Only, and RSA Encrypt-Only keys.  These types are deprecated.  The "key flags" subpacket in a signature is a much better way to express the same idea, and generalizes it to all algorithms.

Also, sort of a different point, but if we look at how RSA is treated in gpg, for example, it defaults to the best practice of generating a primary signing key and an encrypting subkey. So that's the default but it does *permit* you to generate an RSA key that both signs+encrypts. It's up to the user to ultimately decide and accept the risk.

Anyways, I don't have a strong opinion on it, it just sort of jumped out at me as slightly divergent.